cartridges/plugin_beam/cartridge/templates/default/beam/beam_community_impact.isml

```diff
@@ -1,11 +1,24 @@
 <isset value="${dw.system.Site.current.preferences.custom}" name="sitePref" scope="page"/>
-<isset value="${request.getHttpLocale()}" name="currentLocale" scope="page"/>
-<isset value="${currentLocale.split('_')[0]}" name="languageCode" scope="page"/>
-<isset value="${sitePref.beamSupportedLang}" name="supportedLang" scope="page" />
-<isset value="${supportedLang && supportedLang.split(',').indexOf(languageCode) !== -1 ? languageCode : sitePref.beamDefaultLang}" name="finalLanguageCode" scope="page"/>
 
 <script type="module" src="https://sdk.beamimpact.com/web-sdk/${sitePref.beamVersion}/dist/components/community-impact.js"></script>
 
+<isscript>
+    var Site = require('dw/system/Site');
+    var Locale = require('dw/util/Locale');
+    var languageHelper = require('*/cartridge/scripts/utils/getFinalLanguageCode'); // Import the helper function
+
+    var currentLanguageCode = Locale.getLocale(request.locale).getLanguage();
+
+    var finalLanguageCode = languageHelper.getFinalLanguageCode(Site.getCurrent(), currentLanguageCode);
+
+    session.privacy.finalLanguageCode = finalLanguageCode;
```

Comment:
`question` What is `session.privacy.finalLanguageCode`? Is this overwriting something that's built-in to SFCC, or is this just for us?

Reply:
@jmastic-beam it's a custom session variable for our cartridge, so just for us